src/parser.rs

```diff
@@ -1402,6 +1420,101 @@ mod tests {
+                           "k4": b"\x20"
+                         }"#
+                )
+                .unwrap(),
+            HashMap::from([
+                (
+                    "k0".to_owned(),
+                    AttributeValue {
+                        null: Some(true),
+                        ..Default::default()
+                    }
+                ),
+                (
+                    "k1".to_owned(),
+                    AttributeValue {
+                        l: Some(Vec::from([
+                            AttributeValue {
+                                n: Some("1".to_owned()),
+                                ..Default::default()
+                            },
+                            AttributeValue {
+                                n: Some("2".to_owned()),
+                                ..Default::default()
+                            },
+                            AttributeValue {
+                                n: Some("3".to_owned()),
+                                ..Default::default()
+                            },
+                        ])),
+                        ..Default::default()
+                    }
+                ),
+                (
+                    "k2".to_owned(),
+                    AttributeValue {
+                        s: Some("str".to_owned()),
+                        ..Default::default()
+                    }
+                ),
+                (
+                    "k3".to_owned(),
+                    AttributeValue {
+                        m: Some(HashMap::from([
+                            (
+                                "l0".to_owned(),
+                                AttributeValue {
+                                    ns: Some(vec!["1".to_owned(), "2".to_owned()]),
+                                    ..Default::default()
+                                }
+                            ),
+                            (
+                                "l1".to_owned(),
+                                AttributeValue {
+                                    ss: Some(vec!["str1".to_owned(), "str2".to_owned()]),
+                                    ..Default::default()
+                                }
+                            ),
+                            (
+                                "l2".to_owned(),
+                                AttributeValue {
+                                    bool: Some(true),
+                                    ..Default::default()
+                                }
+                            )
+                        ])),
+                        ..Default::default()
+                    }
+                ),
+                (
+                    "k4".to_owned(),
+                    AttributeValue {
+                        b: Some(Bytes::from_static(b"\x20")),
+                        ..Default::default()
+                    }
+                )
+            ])
+        )
+    }
```

Comment:
For the following unit test `test_parse_put_content`, I believe that we can cover all types by just adding the `Number` and `Binary Set`.

If the current code was intentional for something, please let me know.

Reply:
`k1` covers the number type because an element of a List can be any type. However, while writing this, I thought I should include the test case for a mixture of types in a list. I modified this test to cover this scenario.

I also modified it to cover the case for binary set type.

Reply:
I understand that the `Number` has already been covered within a list.
Thank you for adding the test for the mixture of types and the `Binary Set`!